### ISSUE TITLE
UV window suggestions

### ISSUE THREAD

**Issue by iBoonie** [blackears/cyclopsLevelBuilder#59]
Ripping these right from Trenchbroom since they are very nice to have, and are used a lot.

Outlining the used area would be handy (can be seen in any of the videos below, the white outline).
Have the outline be draggable to quick and easily position it wherever you want. 

Being able to have the UV window zoom-able and tiled when you are zoomed out past 100% (have a grid so you can see when it tiles).

https://user-images.githubusercontent.com/62844047/238787160-96fd03a0-a1d6-4fae-94ad-0cbb355fa4ba.mp4

Rotate and change the rotation origin from within the window would be nice. Maybe also have quick buttons to flip on X & Y, and also rotate left & right by 90°?

https://user-images.githubusercontent.com/62844047/a1725685-c189-4f4a-bf74-0b20f9c75d81.mp4

Shearing usually isn't used too too much, but might be a nice addition.

https://user-images.githubusercontent.com/62844047/72d9c9fc-076e-4757-ad8c-fe47d7140964.mp4

Having configurable incremental snapping would be great for [trimsheets](https://www.youtube.com/watch?v=piiYIExIN6E), being able to bump the alignment of the texture with arrow keys would also be extremely nice.

https://user-images.githubusercontent.com/62844047/f2b46548-547c-4cec-9ea1-fcdfdd3b6ce1.mp4

**Comment by elvisish** [comment]
I agree. In my experience, using the UV tool for ground or wall textures has been fine, but trying to align a texture to wholly cover the face of a block (for a tree billboard or such) has been impossible, it's just not easy enough to align the textures precisely at the moment.
